docs/README.md

```diff
@@ -0,0 +1,232 @@
+client that wants to manipulate or retrieve data from a compatible Ed-Fi API
+host.
+
+```mermaid
+flowchart LR
+
+    client(Client App)
+
+    subgraph Data Storage
+        DMS_W[(EdFi_DMS_WRITE)]
+        DMS_R[(EdFi_DMS_READ)]
+        CS_DB[(EdFi_Config)]
+    end
+
+    subgraph Ed-Fi Software
+        DMS[Data Management Service]
+        CS[Configuration Service]
+
+        verb{Verb == GET?}
+
+        DMS -->|GET config| CS
+        CS --> CS_DB
+
+        DMS --> verb
+        verb -->|yes| DMS_R
+        verb -->|no| DMS_W
+    end
+
+    subgraph Third Party Software
+        gateway[API Gateway]
+        openid[OpenID Provider]
+
+        gateway -->|maybe...| openid
+    end
+
+    client -->|/oauth/token| openid
+
+    client -->|/ed-fi/students| gateway
+    gateway -->|/ed-fi/students| DMS
+```
+
+Key components of this system are:
+
+* **Client App**: any application or script that wants to connect to the Ed-Fi
+  API host, written in any programming language. The Client App must request an
+  access token from the OpenID Provider and use that token when issuing requests
+  to the Ed-Fi Data Management Service (via the API Gateway).
+* **API Gateway**: commercial or open-source software for managing and routing
+  HTTP requests. In some situations, the API gateway may be able to use the
+  OpenID provider to check the validity of an access token before
+  forwarding the HTTP request to the Data Management Service.
+* **OpenID Provider**: commercial or open-source software that supports OpenID
+  Connect, which is an extension to the OAuth 2.0 standard.
+* **Data Management Service**: an Ed-Fi API host. It is the direct analogue to
+  the Ed-Fi ODS/API application, which supports HTTP requests based on the Ed-Fi
+  Resources API, Descriptor API, and other API definitions.
+  * **EdFi_DMS_WRITE**: a data store supporting insert, update, and delete operations.
+  * **EdFi_DMS_READ**: a data store support read operations. Could be the same
+    as the "Write" data store or could be in a separate service.
+* **Configuration Service**: a microservice for managing configuration
+  information used to control the Data Management Service at start up.
+  * **EdFi_Config**: a simple database for storing configuration information,
+    including data describing data authorization permissions.
+
+The next diagram is from the perspective of a system administrator interacting
+with this platform to create a new set of client credentials for use by the
+Client App described above.
+
+```mermaid
+flowchart LR
+
+    client(Sys Admin)
+
+    subgraph Ed-Fi Software
+        CS[Configuration Service]
+
+        CS_DB[(EdFi_Config)]
+
+        CS --> CS_DB
+    end
+
+    subgraph Third Party Software
+        gateway[API Gateway]
+        openid[OpenID Provider]
+
+        gateway -->|maybe...| openid
+    end
+
+    client -->|/oauth/token| openid
+
+    client -->|/ed-fi/students| gateway
```

Comment:
The "/ed-fi/students" request endpoint is confusing, since "SysAdmin" try to create new client credentials!